modules/acm/src/main/java/edu/gemini/epics/acm/CaTaskControlImpl.java

```diff
@@ -0,0 +1,477 @@
+        record = new CaTaskControlRecord(channel, epicsService);
+
+        record.registerValListener(valListener = new ChannelListener<Integer>() {
+            @Override
+            public void valueChanged(String arg0, List<Integer> newVals) {
+                if (newVals != null && !newVals.isEmpty()) {
+                    CaTaskControlImpl.this.onValChange(newVals.get(0));
+                }
+            }
+        });
+
+        record.registerStateListener(stateListener = new ChannelListener<TaskControlState>() {
+            @Override
+            public void valueChanged(String arg0, List<TaskControlState> newVals) {
+                if (newVals != null && !newVals.isEmpty()) {
+                    CaTaskControlImpl.this.onStateChange(newVals.get(0));
+                }
+            }
+        });
+
+        executor = new ScheduledThreadPoolExecutor(2);
+
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getApply() {
+        return null;
+    }
+
+    @Override
+    public String getCAR() {
+        return null;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public Set<String> getInfo() {
+        return storage.getInfo();
+    }
+
+    @Override
+    public CaApplySender getApplySender() {
+        return this;
+    }
+
+    @Override
+    public void mark() throws TimeoutException {
+        try {
+            record.setDir(CadDirective.MARK);
+        } catch (CAException e) {
+            LOG.warn(e.getMessage());
+        }
+    }
+
+    @Override
+    public void setTimeout(long timeout, TimeUnit timeUnit) {
+        this.timeout = timeout;
+        this.timeoutUnit = timeUnit;
+    }
+
+    @Override
+    public boolean isActive() {
+            return !currentState.equals(IdleState);
+        }
+
+    @Override
+    public void clear() throws TimeoutException {
+        try {
+            record.setDir(CadDirective.CLEAR);
+        } catch (CAException e) {
+            LOG.warn(e.getMessage());
+        }
+    }
+
+    @Override
+    public CaParameter<Integer> addInteger(String name, String channel, String description, boolean isCADParameter)
+            throws CaException {
+        return storage.addInteger(name, channel, description, isCADParameter);
+    }
+
+    @Override
+    public CaParameter<Double> addDouble(String name, String channel, String description, boolean isCADParameter)
+            throws CaException {
+        return storage.addDouble(name, channel, description, isCADParameter);
+    }
+
+    @Override
+    public CaParameter<Float> addFloat(String name, String channel, String description, boolean isCADParameter)
+            throws CaException {
+        return storage.addFloat(name, channel, description, isCADParameter);
+    }
+
+    @Override
+    public <T extends Enum<T>> CaParameter<T> addEnum(String name, String channel, Class<T> enumType,
```

Comment:
These types of methods are what seem repetitive. Is it not possible to abstract this out to a superclass?